packages/circuit-ui/hooks/useCollapsible/useCollapsible.ts

```diff
@@ -76,18 +76,18 @@ export function useCollapsible<T extends HTMLElement = HTMLElement>({
       duration,
       onStart: () => {
         setHeight(getHeight(contentElement));
-        if (!isOpen) {
-          setOpen(true);
-        }
+        // The timeout forces the state update into the next animation frame.
+        // This ensures that the browsers renders the new height
+        // before the state is toggled.
+        setTimeout(() => {
+          setOpen((prev) => !prev);
+        });
       },
       onEnd: () => {
```

Comment:
It would be nice to have `isOpen` toggle when the animation is done, so we can do things when the animation ends. Or alternatively expose `isAnimating` from the useAnimation hook

Reply:
Actually exposing `isAnimating` might be even better and easier

Reply:
That's a good idea! Done in 834e06a.